### ISSUE TITLE
Need dependencies management, 100 jars imported

### ISSUE THREAD

**Issue by Ikki-Dai** [apache/shardingsphere#20049]
### Is your feature request related to a problem?

Seems shardingsphere 5.x is for plugin? but how to **plug-out** libs that i don't want to import
can provide some advice for maven exclusion, such as i want to exclude 'shadow' or 'schedue'. because i just want simple read-wirte spllitting.

### Describe the feature you would like.

or can provide dependencies management for shardingsphere? and then we can import that we want to use

**Comment by linghengqian** [comment]
- I'm not quite sure how others in the community use it, because in my opinion, the current plugin is mainly for `org.apache.shardingsphere:shardingsphere-jdbc-core:${shardingsphere.version}` SPI mechanism that Maven relies on. expanded. 

- A simple example is that in ShardingSphere Proxy, the additional SPIs you implement can be converted into JARs and then handed over to ShardingSphere Proxy for use, refer to https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/startup/bin/ . 

- Another similar situation is that `org.apache.shardingsphere:shardingsphere-jdbc-core:${shardingsphere.version}` does not contain all the components of the master branch, such as https://shardingsphere.apache.org/document/current/en/user-manual/common-config/builtin-algorithm/sql-translator/ mentioned that the SQL translation module based on JOOQ implementation is not included in this dependency.

**Comment by Ikki-Dai** [comment]
yes, but i also want to import libs 'feature-by-feature', and make it compact in my projects. As an proxy product, it have to import all of libs indeed. 

Can you provide a list and give a short description of each libs?

**Comment by linghengqian** [comment]
Do you want to submit a documentation related PR?  All material is in the developer's manual for the existing documentation.

**Comment by Ikki-Dai** [comment]
i have no idea to simplify the dependencies
while i import `shardingsphere-jdbc-core-spring-boot-starter`, then followed at least **100** jars imported to my project :(

**Comment by linghengqian** [comment]
- The work you requested is likely blocked by https://github.com/apache/shardingsphere/issues/13609.

- I think that the free combination of basic jars is a difficult problem to deal with at present.  Although this is practically feasible, doing this to ShardingSphere JDBC is easily affected by the governance capabilities of ShardingSphere Proxy.  The current ShardingSphere JDBC metadata must be able to provide governance capabilities at the contextmanager level.  For the decoupling of ShardingSphere JDBC and ShardingSphere Proxy, there is https://github.com/apache/shardingsphere/issues/13957 .

- Of course this does not mean that if you need to use JOOQ's SQL translator, you don't need to import `org.apache.shardingsphere:shardingsphere-sql-translator-jooq-provider:5.2.0`.

**Comment by linghengqian** [comment]
- I am working on this issue.  I'll provide a checklist for ShardingSphere Proxy's documentation.

**Comment by Ikki-Dai** [comment]
Glad to see that! hope we will have a bom easier to use

**Comment by linghengqian** [comment]
> hope we will have a bom easier to use

- If you like, open a new issue and try to push it directly. BOM related possibilities are blocked by https://github.com/apache/shardingsphere/pull/17042 .
